### ISSUE TITLE
Migrating suggestions shouldn't fail on duplicates

### ISSUE THREAD

**Issue by dwaynebailey** [translate/pootle#3814]
I get the following when migrating old LO data, we really shouldn't have FATAL ERRORs in issues like this.  We should either fail elegantly or at least do the right thing with a duplicate. This just makes you fearful.

```
 > pootle_store:0009_move_data_from_pootle_app_suggestion
 - Migration 'pootle_store:0009_move_data_from_pootle_app_suggestion' is marked for no-dry-run.
FATAL ERROR - The following SQL query failed:
                                 INSERT INTO pootle_store_suggestion
                                (target_hash, target_f, creation_time, review_time,
                                 user_id, reviewer_id, unit_id, state,
                                 translator_comment_f, translation_project_id)
                                VALUES
                                (%s, %s, %s, %s,
                                 %s, %s, %s, %s,
                                 %s, %s)
The error was: (1062, "Duplicate entry '28723025-4723eb5aa8b0cd28c7e09433839b8fae' for key 'unit_id'"
)
```

**Comment by unho** [comment]
Without a traceback it is difficult to tell where that error is happening.

That SQL query looks like https://github.com/translate/pootle/blob/stable/2.6.0/pootle/apps/pootle_store/migrations/0009_move_data_from_pootle_app_suggestion.py#L116-L124 which is within a `try...except IntegrityError:` so you shouldn't see that error at all.

And can't be https://github.com/translate/pootle/blob/stable/2.6.0/pootle/apps/pootle_store/migrations/0009_move_data_from_pootle_app_suggestion.py#L59 because that shouldn't trigger an `INSERT` since it is a record that already was on the DB.

**Comment by dwaynebailey** [comment]
It's almost definitely the first.   Its the exact same INSERT.  The migration doesn't crash so the fact that it displays is most likely a simple artifact of the situation.

https://github.com/translate/pootle/blob/stable/2.6.0/pootle/apps/pootle_store/migrations/0009_move_data_from_pootle_app_suggestion.py#L134-L135

Seems to indicate that these are indeed handled.  Would be nice to not have the scary noise though.

**Comment by jleclanche** [comment]
Is there a way to reproduce this easily?

**Comment by unho** [comment]
This is fixed in #4064, so closing this manually as this didn't autoclose as usual because the PR base was not master branch.
